python/lsst/daf/butler/core/registry.py

```diff
@@ -57,13 +62,11 @@ def fromConfig(config):
             A new `Registry` subclass instance.
         """
         if not isinstance(config, RegistryConfig):
-            if isinstance(config, str):
-                config = Config(config)
-            if isinstance(config, Config):
-                config = RegistryConfig(config['registry'])
+            if isinstance(config, str) or isinstance(config, Config):
+                config = RegistryConfig(config)
             else:
                 raise ValueError("Incompatible Registry configuration: {}".format(config))
-        cls = doImport(config['cls'])
+        cls = doImport(config['registry.cls'])
```

Comment:
If `config` now is a `RegistryConfig` doesn't it have just the `registry` subset?

Reply:
Ah, I see, because we want the full Butler config for it. Maybe then explicitly specify that?

Reply:
Right. I think it's better for RegistryConfig to only have registry stuff and SchemaConfig to only have schema stuff. I'll change that.